docs/development/testguide.rst

```diff
@@ -148,16 +160,20 @@ Or, to run only the ``wcs`` and ``utils`` tests::
 
     pytest -P wcs,utils
 
-You can also specify a single directory or file to test from the commandline,
-e.g.::
+You can also specify a single directory, file, or even specific test
+to check from the commandline, e.g.::
 
     pytest astropy/modeling
 
 or::
 
     pytest astropy/wcs/tests/test_wcs.py
 
-and this works for ``.rst`` files too::
+or::
+
+    pytest astropy/units/tests/test_quantity.py::TestQuantityCreation::test_float_dtype_promotion
```

Comment:
This is fine but in practice I would always do something shorter like `pytest astropy/units -k float_dtype`. 
Maybe combine all these into examples in a single block of examples:
```
You can also specify a single directory, a file (Python or ``.rst`` doc file), or a specific test
to check from the commandline, e.g.::

    pytest astropy/modeling
    pytest astropy/wcs/tests/test_wcs.py
    pytest astropy/units -k float_dtype_promotion
    pytest astropy/units/tests/test_quantity.py::TestQuantityCreation::test_float_dtype_promotion
    pytest astropy/wcs/index.rst
```

Reply:
The long form has the advantage of tab completion for just the file, and copy & paste from pytest output... But your suggestion of a slew of examples is best!

Reply:
[`pytest` has documentation about test selection](https://docs.pytest.org/en/stable/how-to/usage.html#specifying-which-tests-to-run) and there is no reason we shouldn't link to that and only highlight the differences here. As far as I can tell there are two differences:
1. Thanks to `pytest-doctestplus` it is possible to run tests in docstrings of all `.py` files and tests in all `.rst` files.
2. Thanks to `pytest-filter-subpackage` it is possible to run all tests related to a package (e.g. `wcs`) with `pytest -P wcs`, whereas without the plugin the equivalent invocation would be `pytest {astropy,docs}/wcs`.

Reply:
True, but that the page linked to is substantially longer than the simple list yet does not provide much beyond it. In these development docs, I feel we should just help people get going. 

That said, obviously can include the link...

Reply:
It would also be good to link https://docs.pytest.org/en/stable/how-to/cache.html

Reply:
Thanks! I've pushed an extra commit with both links (learned about `-k` and `--lf` now...). I added `[skip ci]` to the last commit to avoid running all tests again when the docs will do, but obviously fine to squash the commits.